lib/Dialect/FIRRTL/Transforms/LowerXMR.cpp

```diff
@@ -418,6 +453,56 @@ class LowerXMRPass : public LowerXMRBase<LowerXMRPass> {
     return success();
   }
 
+  LogicalResult handlePublicModuleRefPorts(FModuleOp module) {
+    // Decls must be at the top mlir module scope
+    auto declBuilder = ImplicitLocOpBuilder::atBlockBegin(
+        module.getLoc(), getOperation().getBody());
+    auto builder = ImplicitLocOpBuilder::atBlockBegin(module.getLoc(),
+                                                      circuitOp.getBodyBlock());
+
+    for (size_t portIndex = 0, numPorts = module.getNumPorts();
+         portIndex != numPorts; ++portIndex) {
+      auto refType = module.getPortType(portIndex).dyn_cast<RefType>();
+      if (!refType || isZeroWidth(refType.getType()) ||
+          module.getPortDirection(portIndex) != Direction::Out)
+        continue;
+      auto portValue =
+          module.getArgument(portIndex).cast<mlir::TypedValue<RefType>>();
+
+      Attribute ref;
+      SmallString<128> stringLeaf;
+      if (failed(resolveReferencePath(portValue, builder, ref, stringLeaf)))
+        return failure();
+
+      SmallString<128> formatString;
+      if (ref)
+        formatString += "{{0}}";
+      formatString += stringLeaf;
+
+      // Insert a macro with the format:
+      // ref_<circuit name>_<module name>_<ref name> <internal path
+      // from module>
```

Comment:
Nit: the PR could use shorter names to get this into 80-characters:

```suggestion
      //     ref_<circui-namet>_<module-name>_<ref-name> <path>
```

Reply:
Updated